indigo_api/models.py

```diff
@@ -151,6 +151,28 @@ def get_queryset(self):
                             'country__country', 'locality', 'publication_document')
 
 
+class TaxonomyVocabulary(models.Model):
+    authority = models.CharField(max_length=30, null=False, unique=True, blank=False)
+    name = models.CharField(max_length=30, null=False, unique=True, blank=False)
+    slug = models.SlugField(null=False, unique=True, blank=False)
+    title = models.CharField(max_length=30, null=False, unique=True, blank=False)
+
+    def __unicode__(self):
+        return unicode(self.name)
+
+
+class VocabularyTopic(models.Model):
+    taxonomy_vocabulary = models.ForeignKey(TaxonomyVocabulary, related_name='vocabularies', null=False, blank=False, on_delete=models.CASCADE)
+    level_1 = models.CharField(max_length=30, null=False, unique=True, blank=False)
+    level_2 = models.CharField(max_length=30, null=False, unique=True, blank=False)
```

Comment:
level_2 can be null and black, and doesn't need to be unique.

Reply:
level_1 cannot be null and blank?

Reply:
Correct: level_1 is not optional. Level 2 IS optional. Neither need to be unique on their own, only when combined together using the `unique_together` index below.